### PR TITLE
Only remove namespaces that have changed when reloading configs

### PR DESCRIPTION
**Pull request by maniksurtani**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ func (s *server) Start() (bool, error) {
 	}, bufSize)
 
 	<-s.persister.ConfigChangedWatcher()
+	s.createBucketContainer()
```
Comment (kkaneda) [inline server.go]: question: Do we need to call `createBucketContainer` after `<-s.persister.ConfigChangedWatcher()`? Or it won't matter?

Comment (maniksurtani) [inline server.go]: Do you mean every time after `<-s.persister.ConfigChangedWatcher()` and not just here?

File: buckets/bucket_impl_tests.go [hunk 2]
```diff
@@ -63,7 +63,10 @@ func TestGC(t *testing.T, factory quotaservice.BucketFactory, impl string) {
 	helpers.CheckError(t, config.AddNamespace(cfg, nsCfg))
 
 	eventsEmitter := &quotaservice.MockEmitter{Events: make(chan events.Event, 100)}
-	container := quotaservice.NewBucketContainer(cfg, factory, eventsEmitter, quotaservice.NewReaperConfigForTests())
+	container := quotaservice.NewBucketContainer(factory, eventsEmitter, quotaservice.NewReaperConfigForTests())
+	container.Lock()
```
Comment (maniksurtani) [inline buckets/bucket_impl_tests.go]: We technically don't but it's good practice that we do. 

+1 to your suggestion re: `Init()`.

File: server.go (hunk 1 above)
Comment (kkaneda) [inline server.go]: Ah, sorry, no, just wondered the order matters here or not.

Comment (maniksurtani) [inline server.go]: We could create the bucket container before that - not a huge deal either way though.

Comment (maniksurtani) [inline server.go]: May be more readable/obvious though, so you're right.

Comment (kkaneda) [inline server.go]: got it! yeah might be slightly obvious. we could also embed the method to `Start`..

**Pull request merged**